packages/dev/core/src/Engines/WebGL/webGLRenderTargetWrapper.ts

```diff
@@ -102,7 +106,9 @@ export class WebGLRenderTargetWrapper extends RenderTargetWrapper {
             return;
         }
 
-        const framebuffer = this._framebuffer;
+        const mSAARenderBuffer = (texture._hardwareTexture as WebGLHardwareTexture).mSAARenderBuffer;

```

Comment:
I am not too sure why we need this change here ?

I am wondering the diff between _bindTextureRenderTarget( and bindFramebuffer(  and if there should be only one ?

Reply:
this function didn't support the case where the framebuffer is multisampled. If a MSAA render buffer is attached to this renderTarget, the webgl call needs to be  `gl.framebufferRenderBuffer` and not `gl.framebufferTexture2D`.

And to answer your question, both functions have different purposes : 
* `bindTextureRenderTarget` updates the framebuffer with a new texture
* `bindFramebuffer` updates the gl framebuffer pointer for the next draw

Reply:
From what I see:

* _bindTextureRenderTarget will framebufferTexture2D or equivalent to the currently bound FB
* bindFramebuffer first bind an FB, and then also use framebufferTexture2D or equivalent

The second part sounds pretty weird I believe, isnt it @Popov72 ? Maybe it should call _bindTextureRenderTarget  to do so ?

This is probably better suited for another PR but some of the similarities here looks like they could be factorized I guess.

Reply:
it did fail when I started sharing textures amongst framebuffers, let me see if I can make a PG for that